src/main/webapp/js/instructorFeedbackEdit/visibilityOptions.js

```diff
@@ -79,6 +80,20 @@ function attachVisibilityDropdownEvent() {
     });
 }
 
+function checkAndMarkDestructiveChange(selectedOption, $containingForm) {
+    if (selectedOption === 'Custom visibility options...') {
+        return;
+    }
+
+    var currentOption = $containingForm.find('.visibility-options-dropdown button').text();
+    var isSelectionChanged = selectedOption !== currentOption;
+    var hasResponses = $containingForm.attr('editStatus') === 'hasResponses';
+
+    if (isSelectionChanged && hasResponses) {
+        $containingForm.attr('editStatus', 'mustDeleteResponses');
```

Comment:
Does this mean that `mustDeleteResponses`, once set, can never be set back to `hasResponses`? An example will be option A -> option B -> option A, where current choice is option A (same as the beginning, thus no choice) but the flag is set at `mustDeleteResponses`.